lib/src/TailwindComponents/ButtonProposal/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { Button as AriakitButton } from '@ariakit/react'
+
+import { classNames } from '../../classNames'
+
+import buttonStyles from './button.module.scss'
+import type { ButtonProps } from './types'
+
+const cx = classNames(buttonStyles)
+
+// TODO: use forwardRef from system to bring the polymorphism (as props)
```

Comment:
I wouldn't recommand this option, as forwardRef will be deprecated in a future release: https://react.dev/reference/react/forwardRef

I made a PR here to propose a generic type: [bb2fae0](https://github.com/WTTJ/welcome-ui/pull/2792/commits/bb2fae009e5aeefb009c1a7f4f0a5dbc952efc50)